### PR TITLE
Moderate Bugfix 18057

### LINKED ISSUE #13: [18057] Fix the moderate issue
Fix the issue by investigating the bug description, appropriate method and class definitions, and their intended purpose(s). Update the scikit-learn documentation, and write a test case to prove the issue is fixed. Write a technical summary on what was changed in the fix and how the fix will affect the design of scikit-learn. List all the files that were modified.

Time estimation: 6 hours

Acceptance tests:
- [x] run the original use case that displayed the incorrect behavior and show it runs as intended
- [x] edit the docstring documentation for the changed classes and methods to reflect the fixes (including changes to the parameters, behavior, return values, etc.)
- [x] all modified/deleted files are listed in the summary
- [x] a description of how the fix affects scikit-learn
- [x] all modified/deleted files are listed in the summary
- [x] a description of how the fix affects scikit-learn

### PR DESCRIPTION
**Pull request by hemant3434**
#### Reference Issues/PRs

Fixes #18057
#### What does this implement/fix? Explain your changes.
An improved param selector which selects the param dictionary uniformly and makes sure that the bias towards larger dictionaries is removed

#### Any other comments?
Utilizes smart range functions which maintains the current Big O complexity of n**2 on top of added functionality, instead of generating parameters grids every iteration which is the naïve implementation and results in O(n**3).

### CODE REVIEW

**Review by ObaidaSaleh — APPROVED** [review]
LGTM!

**Review by HuIsJason — APPROVED** [review]
LGTM! Hard carry!

**Review by jasoncndai — APPROVED** [review]
LGTM

**Comment by howardleung** [comment]
LGTM 🚀

**Review by howardleung — APPROVED** [review]
LGTM 🚀

**Review by ObaidaSaleh — APPROVED** [review]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -79,6 +79,5 @@ _configtest.o.d
 sklearn/utils/_seq_dataset.pyx
 sklearn/utils/_seq_dataset.pxd
 sklearn/linear_model/_sag_fast.pyx
-
 venv/
-bugs.py
+bugs.py
```
Comment (hemant3434) [inline .gitignore:83]: Must remove before creating a PR to Scikit learn

**Review by swarupsrini — APPROVED** [review]
LGTM

**Pull request closed without merging**